test/integration/01-setup/01-lightning-channels.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { once } from "events"
+import { find } from "lodash"
+import { MainBook } from "src/mongodb"
+import { lndFeePath } from "src/ledger/ledger"
+import {
+  checkIsBalanced,
+  closeChannel,
+  getChannels,
+  getWalletInfo,
+  lnd1,
+  lnd2,
+  lndOutside1,
+  lndOutside2,
+  mineBlockAndSync,
+  openChannelTesting,
+  setChannelFees,
+  subscribeToChannels,
+  subscribeToGraph,
+  updateRoutingFees,
+  waitFor,
+  waitUntilSync,
+} from "test/helpers"
+import { offchainLnds, onChannelUpdated, updateEscrows } from "src/lndUtils"
+import { sleep } from "src/utils"
+
+jest.mock("src/realtimePrice", () => require("test/mocks/realtimePrice"))
+
+//this is the fixed opening and closing channel fee on devnet
+const channelFee = 7637
+const lnds = [lnd1, lnd2, lndOutside1, lndOutside1]
+let channelLengthMain, channelLengthOutside1
+
+beforeEach(async () => {
+  await waitUntilSync({ lnds })
+  channelLengthMain = (await getChannels({ lnd: lnd1 })).channels.length
+  channelLengthOutside1 = (await getChannels({ lnd: lndOutside1 })).channels.length
+})
+
+afterEach(async () => {
+  await checkIsBalanced()
+})
+
+// Setup the next network
+// lnd2 <- lnd1 <-> lndOutside1
+// lnd2 <- lnd1 -> lndOutside2
+// this setup avoids close channels for routing fees tests
+describe("Lightning channels", () => {
+  it("opens channel from lnd1 to lnd2", async () => {
+    const socket = `lnd2:9735`
+    const { lndNewChannel: channel } = await openChannelTesting({
+      lnd: lnd1,
+      lndPartner: lnd2,
+      socket,
+    })
+
+    const { channels } = await getChannels({ lnd: lnd1 })
+    expect(channels.length).toEqual(channelLengthMain + 1)
+
+    await setChannelFees({ lnd: lnd1, channel, base: 0, rate: 0 })
+    await setChannelFees({ lnd: lnd2, channel, base: 0, rate: 0 })
+  })
+
+  it("opens channel from lnd1 to lndOutside1", async () => {
+    const socket = `lnd-outside-1:9735`
+    const { balance: initFeeInLedger } = await MainBook.balance({
+      account: lndFeePath,
+      currency: "BTC",
+    })
+
+    const { lndNewChannel: channel } = await openChannelTesting({
+      lnd: lnd1,
+      lndPartner: lndOutside1,
+      socket,
+    })
+
+    const { channels } = await getChannels({ lnd: lnd1 })
+    expect(channels.length).toEqual(channelLengthMain + 1)
+
+    const { balance: finalFeeInLedger } = await MainBook.balance({
+      account: lndFeePath,
+      currency: "BTC",
+    })
+    expect(finalFeeInLedger - initFeeInLedger).toBe(channelFee * -1)
+
+    await setChannelFees({ lnd: lnd1, channel, base: 1, rate: 0 })
```

Comment:
why do we set the rate at 0 here?

if we want to test how routing fees are affecting our balance sheet we should not have 0 rates.

Reply:
from lnd conf `total fee = basefee + (amount * feerate / 1000000)` so if we set feerate to 0 the total fee will be base fee (it is easier to calc fees in the tests)

ref: https://github.com/lightningnetwork/lnd/blob/master/sample-lnd.conf#L495

Reply:
ok make sense. have a new comment on rounding error in a helper file as a result of this.